source/core_mqtt.c

```diff
@@ -605,7 +605,7 @@ static int32_t sendPacket( MQTTContext_t * pContext,
 
         if( bytesSent < 0 )
         {
-            LogError( ( "Transport send failed. Error code=%d.", bytesSent ) );
+            LogError( ( "Transport send failed. Error code=%d.", ( int ) bytesSent ) );
```

Comment:
Looks like bytesSent is an int32.  For the particular case which started this, the message was
`format '%d' expects argument of type 'int', but argument 4 has type 'int32_t' {aka 'long int'}`

I believe we want to use `%ld` and cast to long to handle the case where int is smaller that 32 bits.

Reply:
Good catch. I'll cast `int32_t` to `long int` and `int16_t` to `int`.

Reply:
Why cast `int16_t` to `int`? A `short` is guaranteed to be at least 2 bytes.

Nit: `long int` and `long` are identical, I think we've been leaving off the `int` in other places where it's not necessary

Reply:
Oh yeah. I think I just wanted to err on the side of caution, but I didn't end up casting any `int16_t` types anyway.